packages/lib-classifier/src/components/Classifier/Classifier.spec.js

```diff
@@ -0,0 +1,431 @@
+
+import RootStore from '@store'
+import { ProjectFactory, SubjectFactory } from '@test/factories'
+import mockStore, { defaultAuthClient, defaultClient } from '@test/mockStore/mockStore'
+import branchingWorkflow from '@test/mockStore/branchingWorkflow'
+import stubPanoptesJs from '@test/stubPanoptesJs'
+import Classifier from './Classifier'
+
+describe('Components > Classifier', function () {
+  let subjectImage, tabPanel, taskAnswers, taskTab, tutorialTab, workflow
+
+  this.timeout(0)
+
+  function withStore(store) {
+    return function Wrapper({ children }) {
+      return (
+        <Grommet theme={zooTheme}>
+          <Provider classifierStore={store}>
+            {children}
+          </Provider>
+        </Grommet>
+      )
+    }
+  }
+
+  describe('while the subject is loading', function () {
+    before(function () {
+      const subject = SubjectFactory.build({ locations: [{ 'image/png': 'https://foo.bar/example.png' }] })
+      const store = mockStore({ subject })
+      const project = store.projects.active
+      const projectSnapshot = { ...getSnapshot(project) }
+      workflow = store.workflows.active
+      const workflowSnapshot = { ...getSnapshot(workflow) }
+      render(
+        <Classifier
+          classifierStore={store}
+          project={projectSnapshot}
+          subjectID={subject?.id}
+          workflowID={workflowSnapshot?.id}
+          workflowSnapshot={workflowSnapshot}
+          workflowVersion={workflowSnapshot?.version}
+        />,
+        {
+          wrapper: withStore(store)
+        }
+      )
+      taskTab = screen.getByRole('tab', { name: 'TaskArea.task'})
+      tutorialTab = screen.getByRole('tab', { name: 'TaskArea.tutorial'})
+      subjectImage = screen.getByRole('img', { name: `Subject ${subject.id}` })
+      tabPanel = screen.getByRole('tabpanel', { name: '1 Tab Contents'})
+      const task = workflowSnapshot.tasks.T0
+      const getAnswerInput = answer => within(tabPanel).getByRole('radio', { name: answer.label })
+      taskAnswers = task.answers.map(getAnswerInput)
+    })
+
+    it('should have a task tab', function () {
+      expect(taskTab).to.be.ok()
+    })
+
+    it('should have a tutorial tab', function () {
+      expect(tutorialTab).to.be.ok()
+    })
+
+    it('should have a subject image', function () {
+      expect(subjectImage).to.be.ok()
+    })
+
+    describe('task answers', function () {
+      it('should be displayed', function () {
+        expect(taskAnswers).to.have.lengthOf(workflow.tasks.T0.answers.length)
+      })
+
+      it('should be linked to the task', function () {
+        taskAnswers.forEach(radioButton => {
+          expect(radioButton.name).to.equal('T0')
+        })
+      })
+
+      it('should be disabled', function () {
+        taskAnswers.forEach(radioButton => {
+          expect(radioButton.disabled).to.be.true()
+        })
+      })
+    })
+  })
+
+  describe('after the subject has loaded', function () {
```

Comment:
Could the "after the subject has loaded" test be in the same test suite as "while the subject is loading"? I think using `rerender` similar to the locale unit test would be closer to how the Classifier behaves in real life. The Classifier component can mount without a subject, and then React re-renders it when `subjectID` prop changes - is that correct?

Reply:
I think that’s right. I think Engaging Crowds projects mount the classifier before you’ve selected a subject to work on, then pass in a subject ID after it has mounted. 

You’re absolutely right about `rerender`. I’m sure there are closed issues because the classifier didn’t originally react properly to prop changes, and we don’t have tests for those.

Reply:
Here's an example of a bug that was caused by a props change.
https://github.com/zooniverse/front-end-monorepo/issues/2140

Reply:
The classifier itself won’t rerender when the subject finishes loading, but the store will update, which then triggers the active task to become enabled. 

The `subjectID` prop is passed down from the page URL, in projects that allow you to select or link to specific subjects.

Reply:
I spent some time working on this today but kept running into RTL’s `afterEach` cleanup. The screen is empty if I go back to examine it again, unless I run a new ‘render’, which is what my existing tests do.